osv/impact_test.py

```diff
@@ -0,0 +1,140 @@
+    affected_commits = affected_commits[0]
+
+    self.assertEqual('BUG-1-0', affected_commits.key.id())
+    self.assertEqual('BUG-1', affected_commits.bug_id)
+    self.assertEqual(0, affected_commits.page)
+    self.assertTrue(affected_commits.public)
+
+    self.assertCountEqual([
+        b'00',
+        b'01',
+        b'02',
+    ], [codecs.encode(commit, 'hex') for commit in affected_commits.commits])
+
+  def test_update_single_page_nonpublic(self):
+    """Test update_affected_commits with a single page."""
+    commits = {
+        '00',
+        '01',
+        '02',
+    }
+
+    impact.update_affected_commits('BUG-1', commits, False, write_legacy=False)
+    affected_commits = list(models.AffectedCommits.query())
+    self.assertEqual(1, len(affected_commits))
+    affected_commits = affected_commits[0]
+
+    self.assertEqual('BUG-1-0', affected_commits.key.id())
+    self.assertEqual('BUG-1', affected_commits.bug_id)
+    self.assertEqual(0, affected_commits.page)
+    self.assertFalse(affected_commits.public)
+
+    self.assertCountEqual([
+        b'00',
+        b'01',
+        b'02',
+    ], [codecs.encode(commit, 'hex') for commit in affected_commits.commits])
+
+  def test_update_multiple_pages(self):
+    """Test update_affected_commits with multiple page."""
+    # These pre-populated pages should be deleted.
+    for i in range(10):
+      models.AffectedCommits(
+          id=f'BUG-1-{i}', bug_id='BUG-1', commits=[], public=True,
+          page=i).put()
+
+    # These should not be deleted as they're for a different bug.
+    models.AffectedCommits(
+        id='BUG-2-0', bug_id='BUG-2', commits=[], public=True, page=0).put()
+
+    commits = {'%08d' % number for number in range(26000)}
+
+    impact.update_affected_commits('BUG-1', commits, True, write_legacy=False)
+    affected_commits = list(models.AffectedCommits.query())
+    self.assertEqual(4, len(affected_commits))
+
+    # Check that the unrelated entry still exists.
+    self.assertTrue(any(c.key.id() == 'BUG-2-0') for c in affected_commits)
+
+    # Check that the new pages got written properly.
+    affected_commits = list(
+        models.AffectedCommits.query(models.AffectedCommits.bug_id == 'BUG-1'))
+
+    for i, result in enumerate(affected_commits):
+      self.assertEqual(f'BUG-1-{i}', result.key.id())
+      self.assertEqual('BUG-1', result.bug_id)
+      self.assertEqual(i, result.page)
+      self.assertTrue(result.public)
+
+    self.assertCountEqual(
+        [b'%08d' % i for i in range(10000)],
+        [codecs.encode(c, 'hex') for c in affected_commits[0].commits])
+    self.assertCountEqual(
+        [b'%08d' % i for i in range(10000, 20000)],
+        [codecs.encode(c, 'hex') for c in affected_commits[1].commits])
+    self.assertCountEqual(
+        [b'%08d' % i for i in range(20000, 26000)],
+        [codecs.encode(c, 'hex') for c in affected_commits[2].commits])
+
+  def test_update_deletes_pages(self):
+    """Test deletion of pages."""
```

Comment:
Unfinished test?

Reply:
Good catch! I didn't need this anymore but I added another test in its place.